<launch>
  <arg name="compute_mode" default="pt_interest"/>
  <!-- URBAN DRIVE -->
  <!-- <arg name="bag" default="/mnt/partition_deux/bag_files/kitti_2011_09_26_drive_0001_synced.bag"/> -->
  <arg name="bag" default="/mnt/partition_deux/bag_files/kitti_2011_09_26_drive_0001_synced.bag"/>
  <!-- <arg name="bag" default="/media/chupin/partition_deux/bag_stephane.bag"/> -->
  

   <!-- <node name="static_tf_lidar2" pkg="tf2_ros" type="static_transform_publisher"
    args="2.0 0.0 1.0  0 0 0  base_link base_link"/> -->

  <!-- <arg name="bag" default="/media/chupin/partition_deux/kitti_dataset/kitti_2011_09_29_drive_0071_synced.bag"/> -->
  <!-- <arg name="bag" default="/media/chupin/partition_deux/kitti_dataset/rough_terrain_1.bag"/> -->
  <!-- PEOPLE DRIVE -->
  <!-- <arg name="bag" default="/media/chupin/partition_deux/kitti_dataset/kitti_2011_09_28_drive_0138_synced.bag"/> -->
  <!-- <node pkg="convert_travel_dataset" type="convert.py" name="convert"/> -->




  <node pkg="big_pc_compute" type="general_compute.py" name="lidar_cloud"
    required="true" output="screen">
    <remap from="~pointcloud" to="/kitti/velo/pointcloud"/>
    <!-- <remap from="~pointcloud" to="/zed2i/zed_node/point_cloud/cloud_registered"/> -->
    <rosparam subst_value="True">
      activate: true
      use_gpu:  true
      debug:    true

      computation_target: $(arg compute_mode)        #["density,"ground_segmentation","pt_interest","experimental_gaussian"]
      frame_id_link: "velo_link"

      density_pcc: 0.2
      density_poi_pcc: 0.5
      density_max_a: -10

      polar_grid: true

      #orthogonnal_grid
      size_voxels: 1.0
      size_voxels_padding: 0.1

      #polar_grid
      angle_voxel_split: 30
      dist_voxel_split:  4.0
      angle_voxel_split_padding: 0.0   #1
      dist_voxel_split_padding:  0.0   #0.2

      pool_n_workers:  4
      pool_n_chunk:    4

      lidar_range_min: 10.0
      lidar_range_max: 40.0
      lidar_z_min:     -2.5
      lidar_z_max:     40.0

      gr_relative_treshold: 0.0
      gr_absolute_treshold: 0.15
      gr_angular:           0.5
      


    </rosparam>
  </node>



  

  <node name="bag_play" pkg="rosbag" type="play" required="true" args="-l $(arg bag)" output="log"/>
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find big_pc_compute)/viz/kitti_v1.rviz" />

</launch>
